tests/x509/verification/test_limbo.py

```diff
@@ -0,0 +1,140 @@
+    "pathlen::max-chain-depth-1-self-issued",
+    # We allow certificates with serial numbers of zero. This is
+    # invalid under RFC 5280 but is widely violated by certs in common
+    # trust stores.
+    "rfc5280::serial::zero",
+    # We allow CAs that don't have AKIs, which is forbidden under
+    # RFC 5280. This is consistent with what Go's crypto/x509 and Rust's
+    # webpki crate do.
+    "rfc5280::ski::root-missing-ski",
+    "rfc5280::ski::intermediate-missing-ski",
+    # We currently allow intermediate CAs that don't have AKIs, which
+    # is technically forbidden under CABF. This is consistent with what
+    # Go's crypto/x509 and Rust's webpki crate do.
+    "rfc5280::aki::intermediate-missing-aki",
+    # We allow root CAs where the AKI and SKI mismatch, which is technically
+    # forbidden under CABF. This is consistent with what
+    # Go's crypto/x509 and Rust's webpki crate do.
+    "webpki::aki::root-with-aki-ski-mismatch",
+}
+
+
+def _get_limbo_peer(expected_peer):
+    kind = expected_peer["kind"]
+    assert kind in ("DNS", "IP")
+    value = expected_peer["value"]
+    if kind == "DNS":
+        return x509.DNSName(value)
+    else:
+        return x509.IPAddress(ipaddress.ip_address(value))
+
+
+def _limbo_testcase(id_, testcase):
+    if id_ in LIMBO_SKIP_TESTCASES:
+        return
+
+    features = testcase["features"]
+    if LIMBO_UNSUPPORTED_FEATURES.intersection(features):
+        return
+    assert testcase["validation_kind"] == "SERVER"
+    assert testcase["signature_algorithms"] == []
+    assert testcase["extended_key_usage"] == [] or testcase[
+        "extended_key_usage"
+    ] == ["serverAuth"]
+    assert testcase["expected_peer_names"] == []
+
+    trusted_certs = [
+        load_pem_x509_certificate(cert.encode())
+        for cert in testcase["trusted_certs"]
+    ]
+    untrusted_intermediates = [
+        load_pem_x509_certificate(cert.encode())
+        for cert in testcase["untrusted_intermediates"]
+    ]
+    peer_certificate = load_pem_x509_certificate(
+        testcase["peer_certificate"].encode()
+    )
+    peer_name = _get_limbo_peer(testcase["expected_peer_name"])
+    validation_time = testcase["validation_time"]
+    validation_time = (
+        datetime.datetime.fromisoformat(validation_time)
+        if validation_time is not None
+        else None
+    )
+    max_chain_depth = testcase["max_chain_depth"]
+    should_pass = testcase["expected_result"] == "SUCCESS"
+
+    verifier = PolicyBuilder(
+        time=validation_time,
+        store=Store(trusted_certs),
+        max_chain_depth=max_chain_depth,
+    ).build_server_verifier(peer_name)
+
+    if should_pass:
+        built_chain = verifier.verify(
+            peer_certificate, untrusted_intermediates
+        )
+
+        # Assert that the verifier returns chains in [EE, ..., TA] order.
+        assert built_chain[0] == peer_certificate
+        assert built_chain[-1] in trusted_certs
```

Comment:
Why no assertion about the intermediates?

Reply:
No good reason, adding.